docs(contributing): Add section for building custom image (#7250)
aws/karpenter-provider-aws: website/content/en/preview/contributing/development-guide.md

make presubmit # run codegen, lint, and tests
```

If you are only interested in building the Karpenter images and not deploying the updated release to your cluster immediately with Helm, you can run the following:

```bash
make image # build and push the karpenter images
```

> *Note: that this will produce a build with the version of https://github.com/kubernetes-sigs/karpenter in your local filesystem.

Author: @engedaam my understanding was that the changes start in preview, then automatically get moved over. It's been quite some time since I submitted the Pr, so let me take a peak at the docs again 😅

> *Note: that this will produce a build with the version of https://github.com/kubernetes-sigs/karpenter in your local filesystem.

Author: Okay I understand what you're suggesting. I have the change in preview, but you'd like me to add it to 
- [v0.32](https://github.com/aws/karpenter-provider-aws/tree/main/website/content/en/v0.32)
- [v0.37](https://github.com/aws/karpenter-provider-aws/tree/main/website/content/en/v0.37)
- [v1.0](https://github.com/aws/karpenter-provider-aws/tree/main/website/content/en/v1.0)
- [v1.1](https://github.com/aws/karpenter-provider-aws/tree/main/website/content/en/v1.1)

correct?

> *Note: that this will produce a build with the version of https://github.com/kubernetes-sigs/karpenter in your local filesystem.

Reviewer: Yeah, we just want to add it to those section

> *Note: that this will produce a build with the version of https://github.com/kubernetes-sigs/karpenter in your local filesystem.

Author: @engedaam sorry, I didn't have a chance to follow up until today. I just pushed a change that rebased off of main since I was quite behind, and then made the updates in all the versions noted above. Let me know if the rebase is fine and if I got the change in all the correct files. Thanks!

> If you are only interested in building the Karpenter images and not deploying the updated release to your cluster immediately with Helm, you can run

Reviewer 2: ```suggestion
If you are only interested in building the Karpenter images and not deploying the updated release to your cluster immediately with Helm, you can run the following:
```

> *Note: that this will produce a build with the latest version of https://github.com/kubernetes-sigs/karpenter.

Reviewer 2: ```suggestion
*Note: that this will produce a build with the version of https://github.com/kubernetes-sigs/karpenter in your local filesystem.
```

```bash

> export $

Reviewer 2: Did you mean to leave something here? Or does this do something I'm not aware of

Author: Ah yeah, I was debating if I wanted to explicitly add setting `$KO_DOCKER_REPO` but that's already called out earlier. Let me update it and remove

> make image # build and push the karpenter images

Reviewer 2: This is automatically done by `make apply`. Was there a reason you wanted to specifically include this as a separate step ?

Author: Same as my comment below. My company is a heavy GitOps shop and I don't deploy karpenter via helm. `make apply` is not relevant to my experience and would actually cause more problems.

```

You can test out changes made in https://github.com/kubernetes-sigs/karpenter by replacing the dependency of https://github.com/aws/karpenter-provider-aws/.
For local changes, replace `$PATH_TO_KUBERNETES_SIGS_KARPENTER` with the relative or absolute path and run:

```bash

> go mod edit -replace sign.k8s.io/karpenter=$PATH_TO_KUBERNETES_SIGS_KARPENTER

Reviewer 2: ```suggestion
go mod edit -replace sigs.k8s.io/karpenter=$PATH_TO_KUBERNETES_SIGS_KARPENTER
```

```

> *Note: you need to commit the go.mod changes before running `make image`

Reviewer 2: you shouldn't need to `git commit` it. If you run `make apply`, it runs the `verify` target, which should run go mod tidy for you

Then you can build your image as before:

```bash
make image # build and push the karpenter images

> ```

Reviewer 2: I think you can probably remove these lines now that you've added more detail above
